### PR TITLE
AA-446: Separate 'nonce' into two 256-bit values 'nonceKey' and 'nonceSequence'

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: contracts/predeploys/NonceManager.sol [hunk 1]
```diff
@@ -26,22 +26,21 @@ contract NonceManager {
...
             let ptr := mload(0x40)
-            calldatacopy(ptr, 0, 44)
+            calldatacopy(ptr, 0, 52)
 
-        // Extract key and sender from calldata
-            let key := shr(64, mload(add(ptr, 20)))
-            mstore(0x00, key)
-            mstore(0x14, shr(96, mload(ptr)))
+        // Extract sender and key from calldata
+            mstore(0x00, shr(96, mload(ptr)))
```
Comment (drortirosh) [inline contracts/predeploys/NonceManager.sol:47]: compeltely unreadable code.
should specify input and output formats.
Honestly, I even think it should use methodSig format..

File: contracts/predeploys/NonceManager.sol [hunk 2]
```diff
@@ -26,22 +33,21 @@ contract NonceManager {
...
+            mstore(0x14, mload(add(ptr, 20)))
 
         // Load nonce from storage
-            nonce := or(shl(64, key), sload(keccak256(0x04, 0x30)))
+            nonce := sload(keccak256(0x00, 0x34))
```
Comment (shahafn) [inline contracts/predeploys/NonceManager.sol]: nonce -> nonceSeq

File: contracts/predeploys/NonceManager.sol [hunk 3]
```diff
@@ -26,22 +33,21 @@ contract NonceManager {
     /// @return nonce a full nonce to pass for next transaction with given sender and key.
```
Comment (shahafn) [inline contracts/predeploys/NonceManager.sol]: Fix comment

File: contracts/predeploys/NonceManager.sol [hunk 4]
```diff
@@ -51,16 +57,16 @@ contract NonceManager {
...
-        // Store key and sender in memory
-            mstore(0x00, shr(64, mload(add(ptr, 20))))
-            mstore(0x14, shr(96, mload(ptr)))
+        // Store sender and key in memory
+            mstore(0x00, shr(96, mload(ptr)))
+            mstore(0x14, mload(add(ptr, 20)))
 
         // Calculate storage slot and load current nonce
-            let nonceSlot := keccak256(0x04, 0x30)
+            let nonceSlot := keccak256(0x00, 0x34)
```
Comment (shahafn) [inline contracts/predeploys/NonceManager.sol]: Fix name, either nonceKey or nonceSeq

**Review by shahafn — APPROVED** [review]

**Pull request merged**